mitodl/ocw-studio
# API endpoint for creating/updating starters from github webhooks

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #291 

#### What's this PR do?
- Adds an API endpoint for processing webhook requests from Github to create or update `WebsiteStarter` objects and their site configs.

#### How should this be manually tested?
- Follow the README section on github integration
- Create a new repo to hold site configs
   
- Use ngrok or something similar to setup a publicly accessible tunnel to your local site:
    - `ngrok http 8043`   
- Create a webhook for the repo (`https://github.com/<org>/<repo>/settings/hooks/`):
   - Payload URL: http://<ngrok_domain>/api/starters/site_configs/
   - Content type: application/json
   - Secret: any value (not blank)
   - Events: pushes
- Set `GITHUB_WEBHOOK_KEY=<secret>` in your `.env` file

- Create new files in the repo with these filepaths:
   ocw-site-1/ocw-studio.yaml
   ocw-site-2/ocw-studio.yaml
   ocw-studio.yaml

   and this content:
   ```
   ---
   collections: []
   ```
- Check that 3 new `WebsiteStarter`  configs were created with a config = `{"collections": []}`
